infobind/cache.go

```diff
@@ -0,0 +1,239 @@
+package infobind
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/parser"
+	"github.com/pingcap/parser/ast"
+	"github.com/pingcap/parser/terror"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/sqlexec"
+	log "github.com/sirupsen/logrus"
+)
+
+//BindData store the basic bind info and bindSql astNode
+type BindData struct {
+	bindRecord
+	ast ast.StmtNode
+}
+
+//BindCache hold a bindDataMap, key:origin sql hash value: bindData slice
+type BindCache struct {
+	Cache map[string][]*BindData
+}
+
+//Handle hold a atomic bindCache
+type Handle struct {
+	bind atomic.Value
+}
+
+//HandleUpdater use to update the BindCache
+type HandleUpdater struct {
+	Parser         *parser.Parser
+	LastUpdateTime types.Time
+	Ctx            sessionctx.Context
+	*Handle
+}
+
+type bindRecord struct {
+	OriginalSQL string
+	BindSQL     string
+	Db          string
+	Status      int64
+	CreateTime  types.Time
```

Comment:
I think it is better to not use `Time`, because we could not guarantee that different clocks in different machines are synced.

Reply:
This time is used to show administrator, so don't need much more precise